structured_type_test.go

```diff
@@ -1621,3 +1623,135 @@ func forAllStructureTypeFormats(dbt *DBTest, f func(t *testing.T, format string)
 		})
 	}
 }
+
+func TestNullObject(t *testing.T) {
+	skipStructuredTypesTestsOnGHActions(t)
+	runDBTest(t, func(dbt *DBTest) {
+		nullObjectQueries := []string{
+			"select null::object(v VARCHAR)",
+			"select null::object",
+		}
+		for i := 0; i < len(nullObjectQueries); i++ {
+			query := nullObjectQueries[i]
+
+			pool := memory.NewCheckedAllocator(memory.DefaultAllocator)
+			defer pool.AssertSize(t, 0)
+			ctx := WithArrowBatches(WithArrowAllocator(context.Background(), pool))
+
+			var err error
+			var rows driver.Rows
+			err = dbt.conn.Raw(func(sc any) error {
+				queryer, implementsQueryContext := sc.(driver.QueryerContext)
+				assertTrueF(t, implementsQueryContext, "snowflake conn does not implement QueryerContext but needs to")
+				rows, err = queryer.QueryContext(ctx, query, nil)
+				return err
+			})
+			assertNilF(t, err, fmt.Sprintf("test failed to run the following query: %s", query))
+			defer rows.Close()
+
+			sfRows, isSfRows := rows.(SnowflakeRows)
+			assertTrueF(t, isSfRows, "rows are not snowflakeRows")
+
+			batches, err := sfRows.GetArrowBatches()
+			assertNilF(t, err)
+			assertNotEqualF(t, len(batches), 0)
+			batch, err := batches[0].Fetch()
+			assertNilF(t, err)
+			assertNotEqualF(t, len(*batch), 0)
+			for _, record := range *batch {
+				// check number of cols/rows so we dont get index out of range
+				assertEqualF(t, record.NumRows(), int64(1), "wrong number of rows")
+				assertEqualF(t, record.NumRows(), int64(1), "wrong number of cols")
+
+				colIndex := 0
+				rowIndex := 0
+
+				srtCol, isStrCol := record.Column(colIndex).(*array.String)
+				assertTrueF(t, isStrCol, "column is not type string")
+
+				assertEqualE(t, srtCol.Value(rowIndex), "")
+				isNull := srtCol.IsNull(rowIndex)
+				assertTrueF(t, isNull)
+				record.Release()
+			}
+		}
+	})
+}
+
+func TestStructuredTypeInArrowBatchesWithSemistructuredObject(t *testing.T) {
```

Comment:
Having a comment below - maybe it is better to call this test `TestSelectingSemistructuredObjectInArrowBatches`? And instead of a duplicated test for structured object - maybe we can have a separate test for semistructured arrays like `SELECT [1, 2, 3]::ARRAY`?

Reply:
I suppose the difference is also that the original tests sets the `ENABLE_STRUCTURED_TYPES_IN_CLIENT_RESPONSE` flag and this test does not. In theory we should be able to just run the new test I added with and without the flag value, but I cant confirm because our test snowflake server wont let us set the flag. Should we combine them? I just cant verify the test runs on my end

Reply:
I do think its good to have `SELECT [1, 2, 3]::ARRAY` and `SELECT {'s':'someString'}::OBJECT` because both of these fail on different branches of `arrowToRecordSingleColumn` (so good call to add this case because I had not fixed it!), are you ok with me leaving both? I think its good to have the higher code coverage

Reply:
Just to explain - this parameter is required for structured types. As long as you use semistructured types (like `::OBJECT` without schema instead of `::OBJECT(v VARCHAR, i INTEGER)` with schema) this parameter does not matter. But I like the tests in current version, so I'm resolving this comment!